src/main/java/io/cryostat/recordings/RecordingMetadataManager.java

```diff
@@ -611,18 +618,32 @@ public Map<String, String> parseRecordingLabels(String labels) throws IllegalArg
         }
     }
 
-    private void transferMetadataIfRestarted(ConnectionDescriptor cd, String oldJvmId) {
+    private void transferMetadataIfRestarted(ConnectionDescriptor cd) {
         try {
             String targetId = cd.getTargetId();
-            String newJvmId = jvmIdHelper.computeJvmId(cd);
-
-            if (newJvmId == null) {
-                logger.info(
-                        "Couldn't generate a new jvmId for target {} with old jvmId {}",
-                        targetId,
-                        oldJvmId);
+            String newJvmId = jvmIdHelper.getJvmId(targetId);
+
+            Path subdirectoryPath = null;
+            for (String encodedJvmId : fs.listDirectoryChildren(archivedRecordingsPath)) {
```

Comment:
Should this be the recordingMetadataDir then? I'm a bit confused on the checking for the connectUrl for the archivedRecordingsPath in the metadataManager if there is already connectUrls in the archivedRecordingsPath we are using as well.

Reply:
That might work, but I wasn't sure if we could always rely on that when this piece of code is running. If there is stored metadata for expected active recordings on a target, but we have never archived anything from that target, would this method be run and would it need to know the JVM ID and connectUrl?

It does seem like duplicated information but I'm not sure if it's always available in both locations, depending on what actions have been performed in the past.

Reply:
If a target was restarted and nothing was ever archived, I don't think this would be run, but I guess that's okay since ActiveRecording metadata shouldn't be saved anyways.

Reply:
Once I figure out this itest failure I'll look at refactoring this to use the existing archives `connectUrl` files.